[None][feat] GPT-OSS Sm120/Sm121 Support

farazkh80 opened the pull request:
<!--
Please write the PR title by following this template:

**[JIRA ticket/NVBugs ID/GitHub issue/None][type] Summary**

Valid ticket formats:
  - JIRA ticket: [TRTLLM-1234] or [FOOBAR-123] for other FOOBAR project
  - NVBugs ID: [https://nvbugs/1234567]
  - GitHub issue: [#1234]
  - No ticket: [None]

Valid types (lowercase): [fix], [feat], [doc], [infra], [chore], etc.

Examples:
  - [TRTLLM-1234][feat] Add new feature
  - [https://nvbugs/1234567][fix] Fix some bugs
  - [#1234][doc] Update documentation
  - [None][chore] Minor clean-up

Alternative (faster) way using CodeRabbit AI:

**[JIRA ticket/NVBugs ID/GitHub issue/None] @coderabbitai title**

NOTE: "@coderabbitai title" will be replaced by the title generated by CodeRabbit AI, that includes the "[type]" and title.
For more info, see /.coderabbit.yaml.

-->

## Description


Adding gpt-oss support for sm120 and SM121. The main changes include
  - Attention sink support in fmha for SM120/121 by @PerkzZheng 
  - Custom cublass GEMM algo optimized for spark by @ttyio
  - Mixed FP8×FP4 CUTLASS Group GEMM by @farazkh80 @djns99 for SM120.



## Test Coverage

### E2E functionality tests

Added
```
test_e2e.py::test_ptp_quickstart_advanced[GPT-OSS-20B-gpt_oss/gpt-oss-20b]
test_e2e.py::test_ptp_quickstart_advanced[GPT-OSS-120B-gpt_oss/gpt-oss-120b]
```

to `tests/integration/test_lists/test-db/l0_rtx_pro_6000.yml` 

### Accuracy tests


Added
```
accuracy/test_llm_api_pytorch.py::TestGPTOSS::test_w4_1gpu[True-True-cutlass-auto]
```

to `tests/integration/test_lists/test-db/l0_rtx_pro_6000.yml` 


Using `trtllm-serve openai/gpt-oss-20b` and ns eval
```
nemo-run_1/0 ----------------------------------------- mmlu ----------------------------------------
nemo-run_1/0 evaluation_mode | num_entries | avg_tokens | gen_seconds | symbolic_correct | no_answer
nemo-run_1/0 pass@1          | 14042       | 132        | 252         | 81.14%           | 0.25%
```


Using `trtllm-serve openai/gpt-oss-120b` and ns eval
```
nemo-run_1/0 ----------------------------------------- mmlu ----------------------------------------
nemo-run_1/0 evaluation_mode | num_entries | avg_tokens | gen_seconds | symbolic_correct | no_answer
nemo-run_1/0 pass@1          | 14042       | 107        | 338         | 85.70%           | 0.13%
```

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- The reviewers assigned automatically/manually are appropriate for the PR.


- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

PerkzZheng committed:
Support attention sinks for ampere-style kernels, fix fp8 fmha, and remove assert

This commit consolidates improvements from three changes:
- Added support for attention sinks in Ampere-style kernels
- Fixed FP8 FMHA kernels
- Removed unnecessary assert

Signed-off-by: Perkz Zheng <67892460+PerkzZheng@users.noreply.github.com>
Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 committed:
add mxfp8mxfp4 kernel gen

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 committed:
fp4 changes

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 committed:
fix generate_kernels

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 committed:
compilation fix

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 committed:
support mixed path in 120 mma and restrict tile shapes to 128x128x128

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 committed:
added fmha fix

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 committed:
fix space and missing trtllm

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 committed:
remove extra comments

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

PerkzZheng reviewed (commented):
@farazkh80 have [those tests](https://github.com/NVIDIA/TensorRT-LLM/blob/main/tests/integration/defs/accuracy/test_llm_api_pytorch.py#L3134) been added automatically to sm120 machines ? the changes in fmha_v2 LGTM.

farazkh80 commented on cpp/tensorrt_llm/kernels/cutlass_kernels/moe_gemm/moe_gemm_template_dispatch.h:
```diff
@@ -616,6 +618,26 @@ MoeGemmRunner<T, WeightType, OutputType, ScaleBiasType>::getTmaWarpSpecializedCo
+        {
+            // Restrict SM120 MXFP8xMXFP4 to 128x128x128 only
+            tma_ws_configs.erase(
+                std::remove_if(
```
agreed, I removed it and also stopped ommiting kernels for MX path with CTA shapes other than 128x128x128 in `generate_kernels.py`.

Moved the filtering to `get_candidate_configs_sm120`.

farazkh80 commented on cpp/tensorrt_llm/kernels/cutlass_kernels/moe_gemm/moe_gemm_template_dispatch.h:
```diff
@@ -616,6 +618,26 @@ MoeGemmRunner<T, WeightType, OutputType, ScaleBiasType>::getTmaWarpSpecializedCo
     }
 
+    // Filter out SM120 mixed FP8xFP4 configs that use K=64 tiles ("...64B")
+    // Prefer K in {128, 256} only for MX path on SM120.
```
removed

farazkh80 commented on cpp/tensorrt_llm/kernels/cutlass_kernels/moe_gemm/moe_gemm_template_dispatch.h:
```diff
@@ -547,13 +548,14 @@ MoeGemmRunner<T, WeightType, OutputType, ScaleBiasType>::getTmaWarpSpecializedCo
-    if ((sm == 120 || sm == 121)
-        && !tensorrt_llm::kernels::cutlass_kernels::isValidSM120MOESpecialisation<T, WeightType>())
+    if ((sm == 120 || sm == 121) && !tensorrt_llm::kernels::cutlass_kernels::isValidSM120MOESpecialisation<T, WeightType>()
+        && !(std::is_same_v<T, __nv_fp8_e4m3> && std::is_same_v<WeightType, __nv_fp4_e2m1>))
```
already had `isValidXXMOESpecialization` edited, so removed redundancy thanks for noticing!

farazkh80 commented on cpp/tensorrt_llm/kernels/cutlass_kernels/moe_gemm/launchers/moe_gemm_tma_ws_launcher.inl:
```diff
@@ -441,7 +448,10 @@ using namespace cutlass::epilogue;
+            /* For SM120 grouped GEMM, use PtrArray schedules. BlockScaled (FP8/FP4) prefers pingpong. */                                                                                                                                                                                                                   \
+            using KernelScheduleSM120 = std::conditional_t<IsBlockScaled,                                                                                                                                                                                                                                                   \
+                cutlass::gemm::KernelPtrArrayTmaWarpSpecializedPingpong,                                                                                                                                                                                                                                                    \
+                cutlass::gemm::KernelPtrArrayTmaWarpSpecializedCooperative>;                                                                                                                                                                                                                                                \
```
fixed this, it was a WAR initially but auto works and stage count auto also works

farazkh80 commented on cpp/tensorrt_llm/kernels/cutlass_kernels/moe_gemm/launchers/moe_gemm_tma_ws_launcher.inl:
```diff
@@ -420,6 +423,10 @@ using namespace cutlass::epilogue;
                 sizeof(typename CollectiveEpilogue::SharedStorage))>;                                                                                                                                                                                                                                                       \
                                                                                                                                                                                                                                                                                                                             \
+            /* SM120 BlockScaled (FP8 x FP4) requires Stages >= 2   */                                                                                                                                                                                                                                                      \
+            using StageCountType = std::conditional_t<IsSM120 && IsBlockScaled,                                                                                                                                                                                                                                             \
```
removed, explained in https://github.com/NVIDIA/TensorRT-LLM/pull/7937/files#r2373980436

farazkh80 committed:
address PR args and clean code

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 commented on cpp/tensorrt_llm/kernels/cutlass_kernels/cutlass_heuristic.cpp:
```diff
@@ -503,8 +503,16 @@ std::vector<CutlassGemmConfig> get_candidate_configs_sm120(CutlassGemmConfig::Ca
     {
         std::vector<CutlassGemmConfig> candidate_configs;
-        if ((config & CutlassGemmConfig::FP4_ONLY) != 0)
+        if (config & CutlassGemmConfig::FP8FP4_MIXED)
```
@djns99  what do you think about a filter like this?

farazkh80 commented on cpp/tensorrt_llm/kernels/cutlass_kernels/moe_gemm/moe_gemm_template_dispatch.h:
```diff
@@ -552,8 +554,7 @@ MoeGemmRunner<T, WeightType, OutputType, ScaleBiasType>::getTmaWarpSpecializedCo
     }
-    if ((sm == 120 || sm == 121)
-        && !tensorrt_llm::kernels::cutlass_kernels::isValidSM120MOESpecialisation<T, WeightType>())
+    if ((sm == 120 || sm == 121) && !tensorrt_llm::kernels::cutlass_kernels::isValidSM120MOESpecialisation<T, WeightType>())
```
will undo this change

farazkh80 commented on cpp/tensorrt_llm/kernels/cutlass_kernels/moe_gemm/moe_gemm_template_dispatch.h:
```diff
@@ -630,11 +631,11 @@ bool MoeGemmRunner<T, WeightType, OutputType, ScaleBiasType>::isTmaWarpSpecializ
 bool MoeGemmRunner<T, WeightType, OutputType, ScaleBiasType>::supportsTmaWarpSpecialized(int sm)
 {
-    return (sm == 90 && tensorrt_llm::kernels::cutlass_kernels::isValidHopperMOESpecialisation<T, WeightType>())
+    return (sm == 90 && kernels::cutlass_kernels::isValidHopperMOESpecialisation<T, WeightType>())
```
undo these all

farazkh80 committed:
remove unneccesaary changes

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 committed:
remove unneccesaary changes 2

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 commented on cpp/tensorrt_llm/kernels/cutlass_kernels/cutlass_heuristic.cpp:
```diff
@@ -515,10 +523,7 @@ std::vector<CutlassGemmConfig> get_candidate_configs_sm120(CutlassGemmConfig::Ca
-        {
-            TLLM_THROW("Not Implemented: SM120 group GEMM only supports nvfp4.");
-        }
+        TLLM_THROW("Not Implemented: SM120 group GEMM only supports nvfp4.");
```
perhaps will change this message

farazkh80 commented on cpp/tensorrt_llm/kernels/cutlass_kernels/cutlass_heuristic.cpp:
```diff
@@ -503,8 +503,16 @@ std::vector<CutlassGemmConfig> get_candidate_configs_sm120(CutlassGemmConfig::Ca
+        }
+        else if (config & CutlassGemmConfig::FP4_ONLY)
+        {
+            // FP4 x FP4: allow all four tiles
```
I think we should remove 128x256x64 as I get a failure
```
[ RUN      ] MixtureOfExpertsTest/2.ConfigSweep
/trtllm-workspace/cpp/tests/unit_tests/kernels/mixtureOfExpertsTest.cu:2248: Failure
Expected: { this->mActType = activation_type; for (auto k : {2, 3}) { this->mOverrideSelectedConfig1 = conf1; this->mOverrideSelectedConfig2 = conf2; this->BasicPermuteTest(k, this->mDeviceMinimumAlignment); if (::testing::Test::HasFailure()) throw std::runtime_error("Test k=" + std::to_string(k) + " Failed"); } } doesn't throw an exception.
  Actual: it throws tensorrt_llm::common::TllmException with description "[TensorRT-LLM][ERROR] Assertion failed: Failed to initialize cutlass TMA WS grouped gemm. Error: Error Internal (/tmp/my_trt_build_using_wheel/tensorrt_llm/kernels/cutlass_kernels/cutlass_instantiations/gemm_grouped/120/cutlass_kernel_file_gemm_grouped_sm120_M128_BS_group2.generated.cu:39)
1       0xbc93b825a088 tensorrt_llm::common::throwRuntimeError(char const*, int, char const*) + 120
2       0xbc93b8b2acdc /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x93acdc) [0xbc93b8b2acdc]
3       0xbc93b8b2b0b4 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x93b0b4) [0xbc93b8b2b0b4]
4       0xbc93b8711f48 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x521f48) [0xbc93b8711f48]
5       0xbc93b8712d90 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x522d90) [0xbc93b8712d90]
6       0xbc93b86ff86c /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x50f86c) [0xbc93b86ff86c]
7       0xbc93b86ffff8 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x50fff8) [0xbc93b86ffff8]
8       0xbc93b8680e48 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x490e48) [0xbc93b8680e48]
9       0xbc93b86e2f48 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x4f2f48) [0xbc93b86e2f48]
10      0xbc93b83fcafc /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x20cafc) [0xbc93b83fcafc]
11      0xbc93b8441728 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x251728) [0xbc93b8441728]
12      0xbc93b8442fd4 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x252fd4) [0xbc93b8442fd4]
13      0xbc93b8538270 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x348270) [0xbc93b8538270]
14      0xbc93b8526704 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x336704) [0xbc93b8526704]
15      0xbc93b8526bf8 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x336bf8) [0xbc93b8526bf8]
16      0xbc93b852717c /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x33717c) [0xbc93b852717c]
17      0xbc93b852e660 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x33e660) [0xbc93b852e660]
18      0xbc93b8527320 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x337320) [0xbc93b8527320]
19      0xbc93b8271d90 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x81d90) [0xbc93b8271d90]
20      0xe603b52a84c4 /usr/lib/aarch64-linux-gnu/libc.so.6(+0x284c4) [0xe603b52a84c4]
21      0xe603b52a8598 __libc_start_main + 152
22      0xbc93b828ee30 /tmp/my_trt_build_using_wheel/tests/unit_tests/kernels/mixtureOfExpertsTest(+0x9ee30) [0xbc93b828ee30]".
Failed
Tactic 1: sm120 tactic with tile shape 128128128 and cluster shape 1001001 dynamic cluster shape 0 mainloop sched 0 epi sched 0 epilogue fusion 0 swap ab 0
Tactic 2: sm120 tactic with tile shape 128256064 and cluster shape 1001001 dynamic cluster shape 0 mainloop sched 0 epi sched 0 epilogue fusion 1 swap ab 0 and activation type: 1
```

farazkh80 committed:
fixed tests detereminsm

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 commented on cpp/tensorrt_llm/kernels/cutlass_kernels/python/generate_kernels.py:
```diff
@@ -904,8 +915,7 @@ def is_mixed_dtype_grouped(op):
             op.gemm_kind == GemmKind.Grouped) and (op.act_type != DataType.e4m3
-                                                   or op.weight_type != e2m1)
-
+                                                   or op.weight_type != e2m1)                             
```
undo

ttyio committed:
Optimize gemm perf for gpt-oss in Spark

1. support bias epilog for cublaslt backend
2. replace torch native mm with trtllm mm for sm121 gpt-oss
3. add lookup table for the gpt-oss gemm for sm121

Signed-off-by: Vincent Huang <vincenth@nvidia.com>
Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 committed:
fix tests for moe

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 commented on cpp/tests/unit_tests/kernels/mixtureOfExpertsTest.cu:
```diff
@@ -222,8 +223,9 @@ protected:
-        bool should_skip_unsupported_fp4 = (getSMVersion() < 100 || getSMVersion() >= 120) && ANY_FP4;
-        return should_skip_no_device || should_skip_unsupported_fp8 || should_skip_unsupported_fp4;
+        bool should_skip_unsupported_fp4 = (getSMVersion() < 100) && ANY_FP4;
+        bool should_skip_unsupported_sm120 = (getSMVersion() >= 120) && !ANY_FPX;
```
fixed

farazkh80 committed:
change error msg

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 committed:
remove comment

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

farazkh80 commented:
/bot run --disable-fail-fast

farazkh80 committed:
pre-commit run

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #20506](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/20506/) [ kill ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #20505](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/20505/) [ run ] completed with state `ABORTED`
<a href="https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/15467/" target="_blank">LLM/main/L0_MergeRequest_PR #15467</a> <a href="https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/15467//display/redirect" target="_blank">(Blue Ocean)</a> completed with status: ABORTED

tensorrt-cicd commented:
[PR_Github #20506](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/20506/) [ kill ] completed with state `SUCCESS`
Successfully killed previous jobs for commit `7150138`

farazkh80 commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #20507](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/20507/) [ run ] completed with state `ABORTED`
<a href="https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/15468/" target="_blank">LLM/main/L0_MergeRequest_PR #15468</a> <a href="https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/15468//display/redirect" target="_blank">(Blue Ocean)</a> completed with status: ABORTED

farazkh80 commented:
/bot run --disable-fail-fast

brb-nv commented on tensorrt_llm/_torch/models/modeling_gpt_oss.py:
```diff
@@ -459,6 +465,11 @@ def __init__(self, model_config: ModelConfig[GptOssConfig]):
         # which may be incompatible with torch.compile due to version mismatch.
         enable_torch_compile_for_embedding = model_config.moe_backend != "TRITON"
 
+        # Use custom cublas since we need LUT to tune the perf.
```
Non-blocking question:
Can you please let me know what LUT means?

farazkh80 replied:
look up table

farazkh80 committed:
add functionality tests

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

brb-nv commented on tensorrt_llm/_torch/models/modeling_gpt_oss.py:
```diff
@@ -43,6 +43,7 @@ def __init__(
         self,
         config: ModelConfig[GptOssConfig],
         layer_idx: int = 0,
+        use_custom_cublas_mm: bool = False,
```
Thanks for letting me know.

brb-nv reviewed (approved):
Changes under `tensorrt_llm/_torch/models/` look good to me.

farazkh80 committed:
lint

Signed-off-by: list <58580514+farazkh80@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #20529](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/20529/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #15487](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/15487/) completed with status: 'SUCCESS'

farazkh80 commented:
/bot run --stage-list "RTXPro6000-PyTorch-[Post-Merge]-1"

tensorrt-cicd commented:
[PR_Github #20578](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/20578/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #15533 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/15533/) completed with status: 'FAILURE'

farazkh80 commented:
/bot run --stage-list "RTXPro6000-PyTorch-Post-Merge-1"

farazkh80 commented:
/bot reuse-pipeline

tensorrt-cicd commented:
[PR_Github #20659](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/20659/) [ reuse-pipeline ] triggered by Bot

mikeiovine reviewed (approved):
Stamping to unblock, @dongfengy's review should be sufficient for this one

voipmonitor commented:
@farazkh80 hello, I have build fresh docker with this merged PR so I can test it on RTX 6000 PRO but I'm getting this error: 

GPUassert: invalid argument /src/tensorrt_llm/cpp/tensorrt_llm/kernels/tinygemm2/tinygemm2_cuda.cu 64

How is it supposed to run please? I'm running it with: 

 trtllm-serve /mnt/gpt-oss-120b --host 0.0.0.0 --port 4997
